Neo4j version 5+ requires a longer password

**Expected Behaviour**
When running a newer image(tag `5.23.0` in my case), the container should start with default settings.

**Actual Behaviour**
```
console.error
    Error: Log stream ended and message "Started." was not received
        at LineStream.<anonymous> (.../node_modules/testcontainers/src/wait-strategies/log-wait-strategy.ts:54:18)
        at LineStream.emit (node:events:517:28)
        at LineStream.emit (node:domain:489:12)
        at endReadableNT (node:internal/streams/readable:1368:12)
        at processTicksAndRejections (node:internal/process/task_queues:82:21)
```

**Testcontainer Logs**
```
  testcontainers [DEBUG] [f7d22df3c0d5] Waiting for log message "Started."... +0ms
  testcontainers [DEBUG] [f7d22df3c0d5] Fetching container logs... +1ms
  testcontainers [DEBUG] [f7d22df3c0d5] Demuxing stream... +2ms
  testcontainers [DEBUG] [f7d22df3c0d5] Demuxed stream +0ms
  testcontainers [DEBUG] [f7d22df3c0d5] Fetched container logs +0ms
  testcontainers:containers [f7d22df3c0d5] Invalid value for password. The minimum password length is 8 characters. +792ms
  testcontainers:containers [f7d22df3c0d5] If Neo4j fails to start, you can: +0ms
  testcontainers:containers [f7d22df3c0d5] 1) Use a stronger password. +0ms
  testcontainers:containers [f7d22df3c0d5] 2) Set configuration dbms.security.auth_minimum_password_length to override the minimum password length requirement. +0ms
  testcontainers:containers [f7d22df3c0d5] 3) Set environment variable NEO4J_dbms_security_auth__minimum__password__length to override the minimum password length requirement. +0ms
  testcontainers:containers [f7d22df3c0d5] org.neo4j.commandline.admin.security.exception.InvalidPasswordException: A password must be at least 8 characters. +1s
  testcontainers [ERROR] [f7d22df3c0d5] Log stream ended and message "Started." was not received +2s
```

**Environment Information**
- Operating System: Macos 14.4
- Docker Version: Engine: 24.0.6
- Node version: 18.18.2
- Testcontainers version: 10.13.0

Will open the PR to fix the default password.

--- a/packages/modules/neo4j/src/neo4j-container.test.ts
+++ b/packages/modules/neo4j/src/neo4j-container.test.ts
@@ -25,6 +25,27 @@ describe("Neo4jContainer", () => {
   });
   // }
 
+  // v5DefaultPassword {
+  it("should connect with custom password", async () => {
+    const container = await new Neo4jContainer("neo4j:5.23.0").start();
+    const driver = neo4j.driver(
+      container.getBoltUri(),
+      neo4j.auth.basic(container.getUsername(), container.getPassword())
+    );
+
+    const session = driver.session();
+    const personName = "Chris";
+    const result = await session.run("CREATE (a:Person {name: $name}) RETURN a", { name: personName });
+    const singleRecord = result.records[0];
+    const node = singleRecord.get(0);
+    expect(node.properties.name).toBe(personName);
+
+    await session.close();
+    await driver.close();
+    await container.stop();
+  });
+  // }
+  
   // setPassword {
   it("should connect with custom password", async () => {
     const container = await new Neo4jContainer().withPassword("xyz1234@!").start();

--- a/packages/modules/neo4j/src/neo4j-container.test.ts
+++ b/packages/modules/neo4j/src/neo4j-container.test.ts
@@ -26,7 +26,7 @@ describe("Neo4jContainer", () => {
   // }
 
   // v5DefaultPassword {
-  it("should connect with custom password", async () => {
+  it("should connect to neo4j:v5 with default password", async () => {
     const container = await new Neo4jContainer("neo4j:5.23.0").start();
     const driver = neo4j.driver(
       container.getBoltUri(),

--- a/packages/modules/neo4j/src/neo4j-container.test.ts
+++ b/packages/modules/neo4j/src/neo4j-container.test.ts
@@ -45,7 +45,7 @@ describe("Neo4jContainer", () => {
     await container.stop();
   });
   // }
-  
+
   // setPassword {
   it("should connect with custom password", async () => {
     const container = await new Neo4jContainer().withPassword("xyz1234@!").start();

--- a/packages/modules/neo4j/src/neo4j-container.ts
+++ b/packages/modules/neo4j/src/neo4j-container.ts
@@ -5,7 +5,7 @@ const HTTP_PORT = 7474;
 const USERNAME = "neo4j";
 
 export class Neo4jContainer extends GenericContainer {
-  private password = "test";
+  private password = "pass123!@#WORD";
   private apoc = false;
   private ttl?: number;
 

